index.js

```diff
@@ -17,41 +17,56 @@ const DEFAULT_ALLOW_HEADERS = [
 
 const DEFAULT_MAX_AGE_SECONDS = 60 * 60 * 24 // 24 hours
 
-const cors = options => handler => (req, res) => {
+const checkOrigin = (origin, allowed) => {
+  if (Array.isArray(allowed)) return !!allowed.find(a => a === origin)
+  if (typeof allowed === 'function') return allowed(origin)
+  if (allowed instanceof RegExp) return allowed.test(origin)
+  return !!allowed
+}
+
+const cors = options => {
   const {
     maxAge,
-    origin,
+    origin = '*',
     allowHeaders,
     allowMethods,
   } = (options || {})
 
-  res.setHeader(
-    'Access-Control-Max-Age',
-    '' + (maxAge || DEFAULT_MAX_AGE_SECONDS)
-  )
+  return handler => (req, res) => {
```

Comment:
A shame that GH couldn't handle the es6 diff here - the LOC changed is not really representative of the changes, but since this was not recognized as equivalent to `options => handler`, this diff looks huge. Apologies, LMK if you'd prefer I broke up these changes.